partitura/io/importmatch.py

```diff
@@ -677,6 +677,14 @@ def part_from_matchfile(
 
             part.add(part_note, onset_divs, offset_divs)
 
+            # Check if the note is tied and if so, add the tie information
+            if is_tied:
+                for el in part.iter_all(end=offset_divs):
+                    if isinstance(el, score.Note):
+                        if el.step == note_attributes["step"] and el.octave == note_attributes["octave"]:
+                            el.tie_next = part_note
+                            part_note.tie_prev = el
+
```

Comment:
nice solution! some minor problems:
- the loop should break when a note is found
- the accidentals should be checked too for pitch matching
- the notes should only be tied if the end/start points match, maybe iter_ending is the way to go
- somehow handle cases without found note (drop note? add? warn?)
- the indentation level seems one too high

Reply:
Thanks for the feedback, points 1,2 are super important I will fix them ASAP
`end=offset_divs` should already cover the third point.
Warning would be the way to go for no note found.

Reply:
For your last point, currently, it only searches for ties when a note is added.